services/web/server/src/simcore_service_webserver/studies_dispatcher/_projects.py

```diff
@@ -0,0 +1,170 @@
+        project_id, "4c69c0ce-00e4-4bd5-9cf0-59b67b3a9343"
+    )
+    viewer_id = compose_uuid_from(project_id, "fc718e5a-bf07-4abe-b526-d9cafd34830c")
+    return file_picker_id, viewer_id
+
+
+def create_viewer_project_model(
+    project_id: str,
+    file_picker_id: str,
+    viewer_id: str,
+    owner: UserInfo,
+    download_link: str,
+    viewer_info: ViewerInfo,
+) -> Project:
+
+    file_picker_output_id = "outFile"
+    file_picker = Node(
+        key="simcore/services/frontend/file-picker",
+        version="1.0.0", # TODO: check with odeimaiz about version here
+        label="File Picker",
+        inputs={},
+        inputNodes=[],
+        outputs={
+            file_picker_output_id: {
+                "downloadLink": download_link,
+                "label": None,
+            }
+        },
+        progress=100,
+    )
+
+    viewer_service = Node(
+        key=viewer_info.key,
+        version=viewer_info.version,
+        label=viewer_info.label,
+        inputs={
+            viewer_info.input_port_key: PortLink(nodeUuid=file_picker_id, output=file_picker_output_id)
+        },
+        inputNodes=[
+            file_picker_id,
+        ],
+    )
+
+    # Access rights policy
+    access_rights = AccessRights(read=True, write=True, delete=True)  # will keep a copy
+    if owner.is_guest:
+        access_rights.write = access_rights.delete = False
+
+    # Assambles project instance
+    project = Project(
+        uuid=project_id,
+        name=f"Viewer {viewer_info.title}",
+        description="Temporary study to visualize downloaded file",
+        thumbnail="https://placeimg.com/171/96/tech/grayscale/?0.jpg",
+        prjOwner=owner.email,
+        accessRights={owner.primary_gid: access_rights},
+        creationDate=now_str(),
+        lastChangeDate=now_str(),
+        workbench={file_picker_id: file_picker, viewer_id: viewer_service},
+        ui=StudyUI(
+            workbench={
+                file_picker_id: {"position": {"x": 305, "y": 229}},
+                viewer_id: {"position": {"x": 633, "y": 229}},
+            }
+        ),
+    )
+
+    return project
+
+
+async def add_new_project(app: web.Application, project: Project, user: UserInfo):
+    # TODO: move this to projects_api
+    # TODO: this piece was taking fromt the end of projects.projects_handlers.create_projects
+
+    from ..projects.projects_db import APP_PROJECT_DBAPI
+    from ..director_v2 import create_or_update_pipeline
+
+    db = app[APP_PROJECT_DBAPI]
+
+    # validated project is transform in dict via json to use only primitive types
+    project_in: Dict = json.loads(project.json())
+
+    # update metadata (uuid, timestamps, ownership) and save
+    _project_db: Dict = await db.add_project(
+        project_in, user.id, force_as_template=False
+    )
+
+    # This is a new project and every new graph needs to be reflected in the pipeline db
+    await create_or_update_pipeline(
+        app, user.id, project["uuid"]
+    )
```

Comment:
I'm a bit astonished you need to call that from here. it should be done in the project_api or so don't you think?

Reply:
I totally agree with you. I added that in a not for further refactoring.
The thing is that we do not have any way in project_api to create a project from scratch and that is in sync with the handlers ... I decided to do it, but in order to avoid possible side effects, i decided to keep it under ``studies_dispatcher._projects`` for the moment. I added that in the comments at the beginning of the function

This will be integrated later in the other module as you suggest.